project/pom.xml

```diff
@@ -86,6 +86,16 @@
   </scm>
 
   <repositories>
+    <repository>
+      <id>gitlab-maven</id>
+      <url>https://gitlab.com/api/v4/projects/19575612/packages/maven</url>
```

Comment:
@nacx will Apache let us depend on a package outside of Maven Central?  This seems dicey in any case.

Reply:
So, first, we need to make sure packages there are ASLv2 compliant, and then yes, we need to make sure we pull dependencies from a source that is guaranteed _to be there's forever. Apache releases must be always buildable from source, and I'm not sure if depending on this would meet these guarantees.
Apart from this, I've already commented on my main concerns in the PR.

Reply:
True, in the long run, and to comply with Apache's terms, specifying a particular repository is not the best choice. Let's use GSON in the original, which is distributed through the standard distribution channel. 

In any case, the consumer is responsible for whether the runtime environment requires a customized bundle or uses the one in the specified repository.

In the context of OSGi, e.g. when creating Karaf assemblies or features, a specific repository can always be specified from where a critical and customized library is loaded.

Translated with www.DeepL.com/Translator (free version)